Add informational logging to GC Queue

tbg opened the pull request:
GC queue runs are the likely culprit of issues during our load tests; this
should give some useful information in the logs when it runs.

Punting on collecting and exposing these stats for now; let's fix the issues
first.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/5282)

<!-- Reviewable:end -->

knz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[storage/gc_queue.go, line 329 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-KCwS8-DDwpfxXcQLZxE:-KCwS8-ESxvJE-EQRMX2:25177540) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc07c84cda3b75be8ae26f0bcca23e07f9bb3ece/storage/gc_queue.go#L329)):*
numIntentsToResolve?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-:-KCwSJhkE9cckwa_HDCy:1417088721)*
<!-- Sent from Reviewable.io -->

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[storage/gc_queue.go, line 258 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-KCwSJ6OBr1T2J7I8S8B:-KCwSJ6OBr1T2J7I8S8C:-1972525533) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc07c84cda3b75be8ae26f0bcca23e07f9bb3ece/storage/gc_queue.go#L258)):*
`s/log.Infof/gcq.eventLog.Infof/g`

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-:-KCwSOlGmV7Y6c8nN15F:1417088721)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[storage/gc_queue.go, line 258 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-KCwSJ6OBr1T2J7I8S8B:-KCwSRrbD3Nq0vvLm3OC:1527727673) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc07c84cda3b75be8ae26f0bcca23e07f9bb3ece/storage/gc_queue.go#L258)):*
You can also get rid of the `gc queue` prefix once you do that.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[storage/gc_queue.go, line 258 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-KCxM1dI1ERQf0l-axG3:-KCxM1dJCv76TECasOEM:482591919) ([raw file](https://github.com/cockroachdb/cockroach/blob/0829d5ac0410173475b153176e3505713e261945/storage/gc_queue.go#L258)):*
are these all really intended to be logged at base verbosity?

---

*[storage/gc_queue.go, line 303 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-KCxM8aXhbAZzWu8D4g0:-KCxM8aXhbAZzWu8D4g1:1525007070) ([raw file](https://github.com/cockroachdb/cockroach/blob/0829d5ac0410173475b153176e3505713e261945/storage/gc_queue.go#L303)):*
probably meant to log `len()` here

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282)*
<!-- Sent from Reviewable.io -->

tbg committed:
Add informational logging to GC Queue

GC queue runs are the likely culprit of issues during our load tests; this
should give some useful information in the logs when it runs.

Punting on collecting and exposing these stats for now; let's fix the issues
first.

tschottdorf commented:
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---

*[storage/gc_queue.go, line 258 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-KCxM1dI1ERQf0l-axG3:-KCxMd9N793EM61uO0-I:1020899730) ([raw file](https://github.com/cockroachdb/cockroach/blob/0829d5ac0410173475b153176e3505713e261945/storage/gc_queue.go#L258)):*
yes. This is code I want to look at tomorrow when the beta cluster crashes.

---

*[storage/gc_queue.go, line 303 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-KCxM8aXhbAZzWu8D4g0:-KCxMlSW6vjuMF4sdL4A:1612494811) ([raw file](https://github.com/cockroachdb/cockroach/blob/0829d5ac0410173475b153176e3505713e261945/storage/gc_queue.go#L303)):*
Yuck, that could've been a horrible log line.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[storage/gc_queue.go, line 258 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-KCxM1dI1ERQf0l-axG3:-KCxNtgeHgBBJ4vFjZhm:1719541337) ([raw file](https://github.com/cockroachdb/cockroach/blob/0829d5ac0410173475b153176e3505713e261945/storage/gc_queue.go#L258)):*
And the beta cluster runs with V(0)?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[storage/gc_queue.go, line 258 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282#-KCxM1dI1ERQf0l-axG3:-KCxSH04YdWSfpCHEVNl:677251276) ([raw file](https://github.com/cockroachdb/cockroach/blob/0829d5ac0410173475b153176e3505713e261945/storage/gc_queue.go#L258)):*
Yes.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5282)*
<!-- Sent from Reviewable.io -->
